src/OrchardCore.Modules/OrchardCore.ContentFields/Settings/ContentPickerFieldSettings.cs

```diff
@@ -0,0 +1,12 @@
+namespace OrchardCore.ContentFields.Settings
+{
+    public class ContentPickerFieldSettings
+    {
+        public string Hint { get; set; }
+        public bool Required { get; set; }
+        public bool Multiple { get; set; }
+        public string[] DisplayedContentTypes { get; set; } = new string[0];
+        public string SearchResultProvider { get; set; }
```

Comment:
I think this should be a custom Editor instead. This way each editor could have its custom settings. We also have the choice to make a default editor resolve all the result providers, that would use some input string as the contract, and a list of available content types to filter on. I will comment separately on the refactoring in another comment.

Reply:
This comment is still valid if you want to solve the Search setting this way. It might also be better to provide custom lookups based on the provider.